### PR TITLE
Fix #493: Fixed pipeline with auth-type auth0

### LINKED ISSUE #493: Fix pipeline with auth-type "auth0"

### PR DESCRIPTION
**Pull request by parthp2107**
### Describe your changes :

Fixed pipelines with auth-type auth0 for security enabled. Fix #493 

### Type of change :

- [x] Bug fix

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [x] I have performed a self-review of my own.
- [x] I have tagged my reviewers below.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

### CODE REVIEW

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/security/JwtFilter.java [hunk 1]
```diff
@@ -119,4 +135,21 @@ protected static String extractToken(MultivaluedMap<String, String> headers) {
...
+
+    // removes white spaces
+    String publicKey = pubKey.replace(" ", "");
+    try {
+      byte[] encode =  Base64.decodeBase64(publicKey);
+      X509EncodedKeySpec keySpecX509 = new X509EncodedKeySpec(encode);
+      KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+      return (RSAPublicKey) keyFactory.generatePublic(keySpecX509);
+    } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
+      e.printStackTrace();
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/security/JwtFilter.java:150]: never print stack trace. User Log.error check other places

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/security/JwtFilter.java [hunk 2]
```diff
@@ -48,13 +54,17 @@
...
+  private static String oktaPublicKey;
 
   @SuppressWarnings("unused")
   private JwtFilter() {
   }
 
   public JwtFilter(AuthenticationConfiguration authenticationConfiguration) {
     this.publicKeyUri = authenticationConfiguration.getPublicKey();
+    this.authProvider = authenticationConfiguration.getProvider();
+    oktaPublicKey = authenticationConfiguration.getOktaPublicKey();
```
Comment (harshach) [inline catalog-rest-service/src/main/java/org/openmetadata/catalog/security/JwtFilter.java:67]: why are we storing this in local field

Comment (harshach) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/security/JwtFilter.java:67]: make it this.oktaPublicKey

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/security/JwtFilter.java (hunk 1 above)
Comment (parthp2107) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/security/JwtFilter.java]: Addressed in the latest commit.

File: catalog-rest-service/src/main/java/org/openmetadata/catalog/security/JwtFilter.java (hunk 2 above)
Comment (parthp2107) [reply catalog-rest-service/src/main/java/org/openmetadata/catalog/security/JwtFilter.java]: Addressed in the latest commit.

**Comment by harshach** [comment]
@parthp2107 did you test these changes with google login?

**Comment by parthp2107** [comment]
Yes @harshach

**Pull request merged**